modules/p2/ding_interests/ding_interests.module

```diff
@@ -0,0 +1,652 @@
+ *   The rating to give the element.
+ *
+ * @return bool
+ *    FALSE if the interest was inserted as a new interest
+ *    TRUE if the interest already existed.
+ */
+function ding_interests_add_to_list($term, $account, $rating) {
+  // Get the historical loan list.
+  $list = ding_list_get_list(DING_INTERESTS_LIST, $account);
+
+  $value = ding_interest_get_term_cql($term);
+
+  if (!ding_list_has_ding_list_element($list, 'taxonomy_term', $value)) {
+    $values = array(
+      'search_title' => $term->name,
+      'value' => $value,
+      'state' => ($rating > DING_INTERESTS_NOT_INTERESTING) ? 'yes' : 'no',
+    );
+    ding_list_create_ding_list_element($list, 'taxonomy_term', $values, $account, TRUE);
+
+    return FALSE;
+  }
+  
+  return TRUE;
+}
+
+/**
+ * Create a ctools ajax button to add or remove elements to the list.
+ *
+ * @term object taxonomy_term
+ *   The taxonomy term being added.
+ *
+ * @account object user
+ *   The user to act on.
+ *
+ * @rating number
+ *   The rating to give the element.
+ *
+ * @remove boolean
+ *   Indicates if this is a remove button or not.
+ *
+ * @return string
+ *   The ajax buttom markup.
+ */
+function ding_interests_ajax_button($term, $account, $rating = DING_INTERESTS_INTERESTING, $remove = FALSE, $titles = array()) {
+  // Include ctools ajax.
+  ctools_include('ajax');
+  $commands = array();
+  // Set ajax button variables.
+  if ($remove) {
+    $path = 'ding/interests/remove/nojs/' . $term->tid . '/' . $account->uid;
+  }
+  else {
+    $path = 'ding/interests/nojs/' . $term->tid . '/' . $account->uid . '/' . $rating;
+  }
+
+  if (!isset($titles['yes'])) {
+    $titles['yes'] = t('Yes! Add to my interests');
+  }
+  if (!isset($titles['no'])) {
+    $titles['no'] = t('No thanks, not interested');
+  }
+
+  // Rating 5 means interested.
+  $title = ($rating == DING_INTERESTS_INTERESTING) ? $titles['yes'] : $titles['no'];
+  $class = ($rating == DING_INTERESTS_INTERESTING) ? "yes" : "no";
+  // Create ajax button.
+  $output = ctools_ajax_text_button($title, $path, $title, "add-to-list transparent-button $class");
+  return $output;
+}
+
+/**
+ * Get interests list elements.
+ *
+ * @param string $type
+ *   The element type.
+ *
+ * @return array
+ *   An array of elements.
+ */
+function ding_interests_get_list_elements($type = FALSE) {
+  $list = ding_list_get_list(DING_INTERESTS_LIST);
+  if (empty($list)) {
+    return array();
+  }
+  $elements = ding_list_load_elements($list);
+  $interests = array();
+  foreach ($elements as $element) {
+    $element_id = ding_list_get_list_element_value($element);
+    if (!$type || ding_list_get_list_element_state($element) == $type) {
+      $interests[] = $element_id;
+    }
+  }
+  return $interests;
+}
+
+/**
+ * Get random interest element.
+ *
+ * @return object
+ *   Taxonomy term corresponding to the interest.
+ */
+function ding_interests_get_random_interest_term() {
+  $term = ding_interests_get_random_list_item();
+  return $term;
```

Comment:
This is a function which does nothing else but call another function and return the output. Is it needed at all?